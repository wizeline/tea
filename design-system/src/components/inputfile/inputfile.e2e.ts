import { newE2EPage } from '@stencil/core/testing';

describe('wz-input-file', () => {
  it('renders an input file component', async () => {
    const page = await newE2EPage();
    await page.setContent('<wz-input-file />');
    const inputFile = await page.find('wz-input-file');
    const label = await inputFile.find('label');
    expect(label.innerText).toEqual('Choose a file');
  });
});
