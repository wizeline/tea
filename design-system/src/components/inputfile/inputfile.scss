wz-input-file {
  display: inline-block;
  height: 40px;
  width: 100%;

  input {
    width: 0.1px;
    height: 0.1px;
    opacity: 0;
    overflow: hidden;
    position: absolute;
    z-index: -1;
  }

  label {
    display: flex;
    align-items: center;
    height: 100%;

    wz-button button {
      border-top-right-radius: 0;
      border-bottom-right-radius: 0;
    }

    span {
      display: flex;
      align-items: center;
      flex: 1;
      color: var(--ink-light);
      cursor: pointer;
      height: 100%;
      background-color: var(--background-base);
      padding: 0 8px;
      min-width: 150px;
      border: solid 1px var(--neutral-dark);
      border-left: none;
      border-top-right-radius: 3px;
      border-bottom-right-radius: 3px;
    }
  }
}
