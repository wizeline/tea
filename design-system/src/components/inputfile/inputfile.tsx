import { Component, Prop, Element, Listen } from '@stencil/core';

@Component({
  tag: 'wz-input-file',
  styleUrl: 'inputfile.scss',
  shadow: false,
})
export class InputFile {
  @Prop()
  name: string = 'file';
  @Prop()
  placeholder: string = 'Choose a file';
  @Prop()
  disabled: boolean = false;
  @Prop()
  accept: string;
  @Element()
  inputFile: HTMLStencilElement;

  label: string;

  @Listen('change')
  todoCompletedHandler(event: CustomEvent) {
    this.label = (event.target as HTMLInputElement).value.split('\\').pop();
    this.inputFile.forceUpdate();
  }

  byPassOnClick() {
    document.getElementById('file').click();
  }

  render() {
    return [
      <input
        type="file"
        name={this.name}
        id="file"
        class="inputfile"
        accept={this.accept}
      />,
      <label htmlFor="file">
        <wz-button onClick={this.byPassOnClick}>{this.placeholder}</wz-button>
        <span>{this.label}</span>
      </label>,
    ];
  }
}
