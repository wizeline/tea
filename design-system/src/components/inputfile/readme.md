# wz-input-file



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description | Type      | Default           |
| ------------- | ------------- | ----------- | --------- | ----------------- |
| `accept`      | `accept`      |             | `string`  | `undefined`       |
| `disabled`    | `disabled`    |             | `boolean` | `false`           |
| `name`        | `name`        |             | `string`  | `'file'`          |
| `placeholder` | `placeholder` |             | `string`  | `'Choose a file'` |


## Dependencies

### Depends on

- [wz-button](../button)

### Graph
```mermaid
graph TD;
  wz-input-file --> wz-button
  style wz-input-file fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
