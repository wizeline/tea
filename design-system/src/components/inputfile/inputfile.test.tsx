import { InputFile } from './inputfile';

import { TestWindow } from '@stencil/core/testing';

describe('wz-input-file', () => {

  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [InputFile],
      html: "<wz-input-file />",
    });
    const input = element.querySelector('input');
    const label = element.querySelector('label');

    expect(input).toBeTruthy();
    expect(label).toBeTruthy();
    expect(label.textContent).toEqual('Choose a file');

    expect(element).toMatchSnapshot();
  });

});
