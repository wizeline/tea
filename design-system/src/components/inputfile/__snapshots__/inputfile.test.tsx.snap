// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-input-file renders value 1`] = `
<wz-input-file
  class="hydrated"
>
  
  <input
    class="inputfile"
    id="file"
    name="file"
    type="file"
  />
  <label
    for="file"
  >
    <wz-button>
      Choose a file
    </wz-button>
    <span>
      
    </span>
  </label>
</wz-input-file>
`;
